src/lib/y2packager/self_update_addon_repo.rb

```diff
@@ -88,8 +88,9 @@ def self.present?(path = REPO_PATH)
     # @return [Boolean] true on success, false if failed
     #
     def self.create_repo(path = REPO_PATH)
-      ret = Yast::Pkg.SourceCreateType("dir://#{URI.escape(path)}?alias=SelfUpdate0",
-        "", "Plaindir")
+      ret = Yast::Pkg.SourceCreateType(
+        "dir://#{Yast::URLRecode.EscapePath(path)}?alias=SelfUpdate0", "", "Plaindir"
```

Comment:
What's the difference to `path.split("/").map { |d| CGI.escape(d) }.join("/")` used below?

Reply:
well, this is dedicated method just for paths. Difference to below method. To be honest I am not sure which one is better as both mangle if it gets non path part of URL and encode it wrongly.

Reply:
also that split is used in test code where I do not want too much rely on other code. But here in "production" code I would like to delete escaping.